rocAL/include/decoders/image/rocjpeg_decoder.h

```diff
@@ -0,0 +1,138 @@
+/*
+Copyright (c) 2025 Advanced Micro Devices, Inc. All rights reserved.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#pragma once
+#include "decoders/image/decoder.h"
+
+#if ENABLE_ROCJPEG
+
+#include "rocjpeg.h"
+
+struct ScalingFactor {
+  unsigned num;
+  unsigned denom;
+};
+
+class HWRocJpegDecoder : public Decoder {
```

Comment:
rocJpeg is hardware Jpeg decoder. So we don't need to explicity name the class HWRocJpegDecoder. RocJpegDecoder is enough

Reply:
There is a class with same name present in rocJpeg library over [here](https://github.com/ROCm/rocJPEG/blob/0442180916e0b1b73bdf10a8864db406aad0eb07/src/rocjpeg_decoder.h#L47)

Using the same name for the class in rocAL resulted in segmentation fault while releasing the handle, as the handle class had a shared ptr for this class. Inorder to resolve this seg fault we had renamed the rocJpeg decoder as such